### PR TITLE
WIP: integrate graphsync for chainsync requests

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by whyrusleeping** [comment]
Current status, I can't quite get the graphsync queries to fetch the data i want them to. Its not finding the msgmeta for some reason?

**Comment by whyrusleeping** [comment]
Reproducing the failures with `go test ./chain/ -v -run TestSyncSimple`

File: chain/blocksync/blocksync_client.go [hunk 1]
```diff
@@ -234,14 +238,45 @@ func (bs *BlockSync) sendRequestToPeer(ctx context.Context, p peer.ID, req *Bloc
...
+	supp, err := bs.host.Peerstore().SupportsProtocols(p, BlockSyncProtocolID, gsproto)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get protocols for peer: %w", err)
+	}
+
+	if len(supp) == 0 {
+		return nil, xerrors.Errorf("peer %s supports no known sync protocols", p)
+	}
+
+	switch supp[0] {
```
Comment (magik6k) [inline chain/blocksync/blocksync_client.go]: Is the order guaranteed to match the one passed to `SupportsProtocols`?

Comment (whyrusleeping) [reply chain/blocksync/blocksync_client.go]: Yeah, Looking at the implementation it returns in the same order.  I also tested it by putting panics in other codepaths.

**Pull request merged**